### PR TITLE
Fix SEACAS deprecated and some compiler warnings

### PR DESCRIPTION
**Pull request by wortiz**
Make compiling new things less annoying, we should add warnings to the jenkins bot so warnings don't build up again

There were also quite a few SEACAS functions that are now deprecated that I updated

**Requires GUTS branch fix-deprecated-and-warnings** as [MP] Weight Function's were being read in improperly and a required field was allowed to be left blank when SUPG was specified

### CODE REVIEW

**Comment by wortiz** [comment]
Theres still some deprecated warnings that need to be fixed so wait on merging this until I get to them.

File: src/ac_particles.c [hunk 1]
```diff
@@ -5639,12 +5639,16 @@ load_restart_file(void)
...
+      char *fgetserr;
       for(i = 0; i < pdim; i++)
 	if(fscanf(fp, "%lf ", &p.x[i]) != 1)
 	  continue;		/* probable EOF */
       if(fscanf(fp, "%lf %d", &p.time, &p.owning_elem_id) != 2)
 	continue;		/* probable EOF */
-      fgets(garbage, 254, fp);
+      fgetserr = fgets(garbage, 254, fp);
+      if (fgetserr == NULL) {
+	EH(-1, "Error reading line in partical restart file");
```
Comment (acochrane) [inline src/ac_particles.c]: Is this supposed to be particle?

File: src/mm_std_models.c [hunk 2]
```diff
@@ -5924,7 +5924,13 @@ electrolyte_temperature (double t,         /* present value of time */
           t0 = tran->init_time;
         }
 #ifndef tflop
-            if(t <= t0) system("rm T_vs_t.out");  /* zero the file before writing to it */ 
+      if(t <= t0)
+	{
+	  int err;
+	  err = system("rm T_vs_t.out");  /* zero the file before writing to it */
```
Comment (acochrane) [inline src/mm_std_models.c]: This seems uncomfortable. Does goma use system for removing other files?

File: src/ac_particles.c (hunk 1 above)
Comment (wortiz) [reply src/ac_particles.c]: Ha, yes.

File: src/mm_std_models.c (hunk 2 above)
Comment (wortiz) [reply src/mm_std_models.c]: There are a couple places that goma uses system when it probably shouldn't

This one can probably be replaced with fopen write vs append

**Comment by wortiz** [comment]
All the deprecated functions should be fixed with the last commit on this, should be ready to merge once it's been looked at

**Comment by acochrane** [comment]
.runguts

**Comment by acochrane** [comment]
This wraps up issue #149 
Thanks @wortiz !

**Pull request merged**